Source/track.cpp

```diff
@@ -111,11 +109,8 @@ void RepeatMouseAction()
 			CheckPlrSpell(false);
 		break;
 	case MouseActionType::OperateObject:
-		if (pcursobj != -1) {
-			auto &object = Objects[pcursobj];
-			if (object.IsDoor())
-				break;
-			NetSendCmdLocParam1(true, CMD_OPOBJXY, object.position, pcursobj);
+		if (ObjectUnderCursor != nullptr && !ObjectUnderCursor->IsDoor()) {
+			NetSendCmdLocParam1(true, CMD_OPOBJXY, ObjectUnderCursor->position, ObjectUnderCursor->GetId());

```

Comment:
Shouldn't this be changed to:
```suggestion
			NetSendCmdLoc(MyPlayerId, true, CMD_OPOBJXY, ObjectUnderCursor->position);
```